code/modules/library/library_equipment.dm

```diff
@@ -0,0 +1,426 @@
+	if(!I.use_tool(src, user, 20, volume = I.tool_volume))
+		return
+	TOOL_DISMANTLE_SUCCESS_MESSAGE
+	deconstruct(TRUE)
+
+/obj/structure/bookcase/wrench_act(mob/user, obj/item/I)
+	. = TRUE
+
+	if(!I.use_tool(src, user, 0, volume = I.tool_volume))
+		return
+	anchored = !anchored
+	to_chat(user, "<span class='notice'>You [anchored ? "wrench" : "unwrench"] [src].</span>")
+
+/obj/structure/bookcase/manuals/medical
+	name = "Medical Manuals bookcase"
+
+/obj/structure/bookcase/manuals/medical/Initialize()
+	. = ..()
+	new /obj/item/book/manual/medical_cloning(src)
+	update_icon()
+
+
+/obj/structure/bookcase/manuals/engineering
+	name = "Engineering Manuals bookcase"
+
+/obj/structure/bookcase/manuals/engineering/Initialize()
+	. = ..()
+	new /obj/item/book/manual/wiki/engineering_construction(src)
+	new /obj/item/book/manual/engineering_particle_accelerator(src)
+	new /obj/item/book/manual/wiki/hacking(src)
+	new /obj/item/book/manual/wiki/engineering_guide(src)
+	new /obj/item/book/manual/engineering_singularity_safety(src)
+	new /obj/item/book/manual/wiki/robotics_cyborgs(src)
+	update_icon()
+
+/obj/structure/bookcase/manuals/research_and_development
+	name = "R&D Manuals bookcase"
+
+/obj/structure/bookcase/manuals/research_and_development/Initialize()
+	. = ..()
+	new /obj/item/book/manual/research_and_development(src)
+	update_icon()
+
+/obj/structure/bookcase/sop
+	name = "bookcase (Standard Operating Procedures)"
+
+/obj/structure/bookcase/sop/Initialize()
+	. = ..()
+	new /obj/item/book/manual/wiki/sop_command(src)
+	new /obj/item/book/manual/wiki/sop_engineering(src)
+	new /obj/item/book/manual/wiki/sop_general(src)
+	new /obj/item/book/manual/wiki/sop_legal(src)
+	new /obj/item/book/manual/wiki/sop_medical(src)
+	new /obj/item/book/manual/wiki/sop_science(src)
+	new /obj/item/book/manual/wiki/sop_security(src)
+	new /obj/item/book/manual/wiki/sop_service(src)
+	new /obj/item/book/manual/wiki/sop_supply(src)
+	update_icon()
+
+/obj/structure/bookcase/random
+	var/category = null
+	var/book_count = 5
+	icon_state = "random_bookcase"
+	anchored = TRUE
+
+/obj/structure/bookcase/random/Initialize(mapload)
+	. = ..()
+	var/list/books = GLOB.library_catalog.get_random_book(book_count, doAsync = FALSE)
+	for(var/datum/cachedbook/book as anything in books)
+		new /obj/item/book(src, book, TRUE, FALSE)
+	update_icon()
+
+/*
+ * Book binder
+ */
+/obj/machinery/bookbinder
+	name = "Book Binder"
+	icon = 'icons/obj/library.dmi'
+	icon_state = "binder"
+	anchored = TRUE
+	density = TRUE
+
+	var/datum/cachedbook/selected_content = new()
+	var/printing = FALSE
+
+/obj/machinery/bookbinder/attack_ai(mob/user)
+	return attack_hand(user)
+
+/obj/machinery/bookbinder/attack_hand(mob/user)
+	if(..())
+		return
+	ui_interact(user)
+
+/obj/machinery/bookbinder/attack_ghost(mob/user)
+	ui_interact(user)
+
+/obj/machinery/bookbinder/attackby(obj/item/I, mob/user)
+	if(default_unfasten_wrench(user, I))
+		power_change()
+		return
```

Comment:
Best to put this in the `wrench_act` proc override for this object

Reply:
implemented